parlai/tasks/msc/agents.py

```diff
@@ -616,3 +639,125 @@ def __init__(self, opt, shared=None):
 
 class DefaultTeacher(MscTeacher):
     pass
+
+
+class ContextGenerator(BaseContextGenerator):
+    """
+    Generates contexts shown to bots for generating prompt when collecting human-human
+    followup chat in the personal knowledge human evaluation.
+
+    This generator was used to generate the context information shown to bots at the
+    beginning of a conversation, when crowdsourcing the conversations that for per-turn
+    human evaluation.
+    """
+
+    def __init__(self, override_opt, datatype='valid', seed: Optional[int] = None):
+        """
+        Initalize the context generator.
+
+        opt: only a 'datapath' key is required, to specify the ParlAI data folder
+        """
+
+        def setup_opt(opt):
+            parser = dsd.setup_args()
+            parser.set_params(**opt)
+            return parser.parse_args([])
+
+        if seed is not None:
+            self.rng = random.Random(seed)
+        else:
+            self.rng = random.Random()
+
+        with open(override_opt['override_opt_path']) as f:
+            override_opt.update(json.load(f))
+
+        bot_model_name = override_opt['bot_model_name']
+        bot_msc_opt = copy.deepcopy(COMMON_CONFIG)
+        bot_msc_opt.update(MODEL_OPT[bot_model_name])
+        ui_msc_opt = copy.deepcopy(COMMON_CONFIG)
+        ui_msc_opt.update(UI_OPT[bot_model_name])
+
+        self.ui_msc_teacher = SessionBaseMscTeacher(setup_opt(ui_msc_opt))
+        self.bot_msc_teacher = SessionBaseMscTeacher(setup_opt(bot_msc_opt))
+        self.bot_sorted_initial_data_indices_to_episode = {}
+        self.ui_sorted_initial_data_indices_to_episode = {}
+        self.initial_data_indices_to_complete = override_opt.get(
+            'initial_data_indices_to_complete', INITIAL_DATA_TO_COMPLETE
+        )
+        self._set_teacher_data_map()
+        self.context_done_statistics = copy.deepcopy(
+            override_opt.get('context_done_statistics', {})
+        )
+
+    def get_context(self, model_name: str = None) -> dict:
+        """
+        Get context information to be shown at the beginning of one conversation.
+
+        Values in return dict:
```

Comment:
nit: it looks like there are more values in the return dict than just these, right? For instance, I see that `._extract_personas() ` returns `"context_for_bot_prompt"`, etc. Maybe it's worth just referring to the subclass's docstring instead of copying it here?

Reply:
correct! sry will update the doc string